modules/homa-lite/src/lib.rs

```diff
@@ -0,0 +1,257 @@
+		#[pallet::weight(0)]
+		#[transactional]
+		pub fn request_mint(origin: OriginFor<T>, amount: Balance) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+			let current_batch = Self::current_batch();
+			let staking_currency_id = T::StakingCurrencyId::get();
+
+			let stash_account = &Self::relaychain_stash_account().ok_or(Error::<T>::RelaychainStashAccountNotSet)?;
+
+			// TODO: Cross-chain transfer to the Relaychain via XCM
+			T::Currency::transfer(staking_currency_id, &who, &stash_account.clone(), amount)?;
+
+			let current = PendingAmount::<T>::get(current_batch, &who);
+			// Due to the math on Total Issuance, it is literally impossible to overflow here.
+			let new_value = current + amount;
+			PendingAmount::<T>::insert(current_batch, &who, new_value);
+
+			Self::deposit_event(Event::<T>::MintRequested(current_batch, who, amount));
+			Ok(().into())
+		}
+
+		/// Process a batch.
+		/// It is then that we can issue Liquid currencies.
+		///
+		/// Parameters:
+		/// - `staking_total_issuance`:
+		//#[pallet::weight(< T as Config >::WeightInfo::issue())]
+		#[pallet::weight(0)]
+		#[transactional]
+		pub fn issue(origin: OriginFor<T>, staking_total_issuance: Balance) -> DispatchResultWithPostInfo {
+			T::IssuerOrigin::ensure_origin(origin)?;
+			let current_batch = Self::current_batch();
+
+			ensure!(
+				staking_total_issuance > 0,
+				Error::<T>::InvalidStakedCurrencyTotalIssuance
+			);
+
+			let total_liquid_issuance = T::Currency::total_issuance(T::LiquidCurrencyId::get());
+			let total_for_batch = TotalIssuanceInfo {
+				staking_total: staking_total_issuance,
+				liquid_total: total_liquid_issuance,
+			};
+
+			BatchTotalIssuanceInfo::<T>::insert(&current_batch, total_for_batch);
+			CurrentBatch::<T>::put(current_batch + 1);
+
+			Self::deposit_event(Event::<T>::BatchProcessed(
+				current_batch,
+				staking_total_issuance,
+				total_liquid_issuance,
+			));
+
+			Ok(().into())
+		}
+
+		/// A function that allows the user to claim the Liquid currencies minted.
+		/// The amount of liquid currency minted is
+		///
+		/// Parameters:
+		/// - `who`: The user the claimed Liquid currency is for.
+		/// - `batch`: The batch index the user Staked their tokens.
+		//#[pallet::weight(< T as Config >::WeightInfo::claim())]
+		#[pallet::weight(0)]
+		#[transactional]
+		pub fn claim(origin: OriginFor<T>, who: T::AccountId, batch: EraIndex) -> DispatchResultWithPostInfo {
+			ensure_signed(origin)?;
+			let staked_amount = PendingAmount::<T>::get(&batch, &who);
+			let total_info =
+				Self::batch_total_issuance_info(batch).ok_or(Error::<T>::LiquidCurrencyNotIssuedForThisBatch)?;
+
+			let mul_res = staked_amount
+				.checked_mul(total_info.liquid_total)
+				.ok_or(ArithmeticError::Overflow)?;
+			let liquid_res = mul_res
+				.checked_div(total_info.staking_total)
+				.ok_or(ArithmeticError::Overflow)?;
+			let liquid_to_mint = liquid_res;
```

Comment:
Multiplying two reasonable balance values almost always overflow. Use `Permill`to get the `liquid_total / staking_total`, and then multiply it with `staked_amount` would work.